### ISSUE TITLE
2nd Turn Not Ending for 'middle' player

### ISSUE THREAD

**Issue by taetem** [cs340tabyu/cs340Winter2017#293]
This bug occurred after giving each player a different color upon joining the game. The game player Brooke joined, exited and re-joined three times. Mark's color on the map is green, but in this client is brown (as seen by the gameplay client). 

I would first try to reproduce this bug before fixing it - I think it was caused by the joining and rejoining of the client, but the color differences made it difficult to track down.

It seems that after Mark's 2nd turn, his turn never finished, and no settlement or road can be placed. Unfortunately, this bug makes it impossible to proceed with gameplay.
![screen shot 2016-10-26 at 6 56 48 pm](https://cloud.githubusercontent.com/assets/15018401/19750707/b78b7518-9bb0-11e6-8fc1-a53323b677ef.png)

**Comment by mclements37** [comment]
@taetem Hey Taetem! We're having a really hard time trying to reproduce this error. So is the screen shot of Mark's client view, or Pete's client view? We're noticing on the turn tracker it is saying its Pete's turn, but you said Mark's turn never finished... so is the screen shot showing mark's client view?

**Comment by taetem** [comment]
Sorry, it looks like you're right! This is during Pete's turn, which is why it makes no sense that he cant place anything. I did notice that there were two overlays on the clients right before this happened. Sorry you haven't been able to replicate it - I think it could have been a case of having two clients for Pete opened, and so only one works. I closed the 'extra' Pete client, and then this happened. (Perhaps just label this bug as a 'feature'? :P ) As it only happens under very specific conditions, it probably isn't likely to break your game - only when you have a crazy person opening client after client and logging in under the same username.
